sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -161,6 +138,17 @@
         // finds any existing configuration settings, then its e-tag will match and the service will return an error.
         return service.setKey(serviceEndpoint, setting.getKey(), setting.getLabel(), setting, null,
             getETagValue(ETAG_ANY), context)
+            .onErrorResume((Function<Throwable, Mono<Response<ConfigurationSetting>>>) throwable -> {
```

Comment:
Use the overload where you match the class HttpResponse, so you don't bother doing this check.
https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html#onErrorResume-java.lang.Class-java.util.function.Function-

Reply:
updated